modules/openapi-generator/src/main/java/org/openapitools/codegen/config/GlobalSettings.java

```diff
@@ -34,7 +34,9 @@
     private static ThreadLocal<Properties> properties = new InheritableThreadLocal<Properties>() {
         @Override
         protected Properties initialValue() {
-            return (Properties) System.getProperties().clone();
+            Properties copy = new Properties();
```

Comment:
Thanks for the PR. Can you please add a 1-liner comment explaining why and mentioning https://github.com/gradle/gradle/issues/17344 ?

Reply:
Done. Thanks for reviewing!